pkg/components/velero/openebs/openebs.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package openebs deals with configuring Velero openebs plugin.
+package openebs
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/kinvolk/lokomotive/internal"
+)
+
+const indentation = 6
+
+// Configuration contains OpenEBS specific parameters.
+type Configuration struct {
+	Credentials            string                  `hcl:"credentials"`
+	BackupStorageLocation  *BackupStorageLocation  `hcl:"backup_storage_location,block"`
+	VolumeSnapshotLocation *VolumeSnapshotLocation `hcl:"volume_snapshot_location,block"`
+	CredentialsIndented    string
+}
+
+// BackupStorageLocation configures the backup storage location for OpenEBS plugin.
+type BackupStorageLocation struct {
+	Bucket   string `hcl:"bucket"`
+	Region   string `hcl:"region"`
+	Provider string `hcl:"provider,optional"`
+	Name     string `hcl:"name,optional"`
+}
+
+// VolumeSnapshotLocation configures the volume snapshot location for OpenEBS plugin.
+type VolumeSnapshotLocation struct {
+	Bucket           string `hcl:"bucket"`
+	Region           string `hcl:"region"`
+	Provider         string `hcl:"provider,optional"`
+	Name             string `hcl:"name,optional"`
+	Prefix           string `hcl:"prefix,optional"`
+	Local            bool   `hcl:"local,optional"`
+	OpenEBSNamespace string `hcl:"openebs_namespace,optional"`
+	S3URL            string `hcl:"s3_url,optional"`
+}
+
+// NewConfiguration returns the new default Configuration.
+func NewConfiguration() *Configuration {
+	return &Configuration{
+		BackupStorageLocation: &BackupStorageLocation{
+			Provider: "aws",
+			Name:     "default",
+		},
+		VolumeSnapshotLocation: &VolumeSnapshotLocation{
+			Provider:         "aws",
+			Name:             "default",
+			Prefix:           "cstor",
+			OpenEBSNamespace: "openebs",
+		},
+	}
+}
+
+// ChartValuesTemplate returns the chart values template.
+func (c *Configuration) ChartValuesTemplate() string {
+	return chartValuesTmpl
+}
+
+// IndentCredentials indents the provided credentials.
+func (c *Configuration) IndentCredentials() {
```

Comment:
Hm, to avoid adding this function to the interface (it seems nasty), Perhaps we should change `ChartValuesTemplate` to `Values() (string, error)` or `ChartValues() (string, error)` so we can indent as part of the rendering?